app/src/ui/history/compare.tsx

```diff
@@ -401,35 +401,34 @@ export class CompareSidebar extends React.Component<
         event.preventDefault()
         return
       }
-
-      if (this.props.compareState.filterText.length === 0) {
-        this.handleEscape()
+      if (this.state.focusedBranch == null) {
+        this.viewHistoryForBranch()
       } else {
-        if (this.state.focusedBranch == null) {
-          this.viewHistoryForBranch()
-        } else {
-          const branch = this.state.focusedBranch
-
-          this.props.dispatcher.executeCompare(this.props.repository, {
-            kind: HistoryTabMode.Compare,
-            comparisonMode: ComparisonMode.Behind,
-            branch,
-          })
-
-          this.props.dispatcher.updateCompareForm(this.props.repository, {
-            filterText: branch.name,
-          })
-        }
+        const branch = this.state.focusedBranch
```

Comment:
:art: extract this variable outside of the `if` statement, so this becomes a bit more succinct:

```diff
diff --git a/app/src/ui/history/compare.tsx b/app/src/ui/history/compare.tsx
index 7fa6d96f5..0b8be3f7e 100644
--- a/app/src/ui/history/compare.tsx
+++ b/app/src/ui/history/compare.tsx
@@ -401,11 +401,11 @@ export class CompareSidebar extends React.Component<
         event.preventDefault()
         return
       }
-      if (this.state.focusedBranch == null) {
+      const branch = this.state.focusedBranch
+
+      if (branch == null) {
         this.viewHistoryForBranch()
       } else {
-        const branch = this.state.focusedBranch
-
         this.props.dispatcher.executeCompare(this.props.repository, {
           kind: HistoryTabMode.Compare,
           comparisonMode: ComparisonMode.Behind,
```

Reply:
Consider it done!